src/Spice86.Core/Emulator/CPU/CfgCpu/ExecutionContextManager.cs

```diff
@@ -9,10 +9,10 @@ public class ExecutionContextManager {
     private MachineBreakpoints _machineBreakpoints;
     private IDictionary<uint, ExecutionContext> _executionContextEntryPoints = new Dictionary<uint, ExecutionContext>();
     
-    public ExecutionContextManager(MachineBreakpoints machineBreakpoints) {
+    public ExecutionContextManager(MachineBreakpoints machineBreakpoints, ExecutionContext executionContext) {
         _machineBreakpoints = machineBreakpoints;
         // Initial context at init
-        CurrentExecutionContext = new();
+        CurrentExecutionContext = executionContext;
```

Comment:
I'm not sure it is worth passing this in the constructor as the next contexts are created via new anyway.

Reply:
If it's not injected, I can't eventually remove the UI visitor in the executionContext class:

     public void Accept<T>(T emulatorDebugger) where T : IInternalDebugger {
        emulatorDebugger.Visit(this);
     }

and the complex way of initializing the internal debugger viewModels on the UI side.

I'll try another solution, as the new in SignalNewExecutionContext also defeats the whole point of IoC.

Reply:
I removed the injection of the initial ExecutionContext in the ExecutionContextManager. The ExecutionContextManager will be injected in the CfgCpuViewModel, instead of it visiting the ExecutionContext.